### PR TITLE
Add associated model id to mesh collection

### PR DESCRIPTION
**Pull request by yumin**

### CODE REVIEW

**Comment by yumin** [comment]
@robertmaynard comments?

File: smtk/mesh/Collection.cxx [hunk 1]
```diff
@@ -502,5 +502,33 @@ bool Collection::setNeumannOnMeshes(const smtk::mesh::MeshSet& meshes,
...
 
+//----------------------------------------------------------------------------
+bool Collection::associateModel(const smtk::common::UUID& uuid)
+{
+  if(this->m_modelEntity == uuid)
+    {
+    return true;
+    }
+
+  if(!this->m_modelEntity.isNull() && uuid.isNull())
```
Comment (robertmaynard) [inline smtk/mesh/Collection.cxx]: this can be removed ```setModelEntity``` returns false if uuid is null.

File: smtk/mesh/Collection.cxx [hunk 2]
```diff
@@ -502,5 +502,33 @@ bool Collection::setNeumannOnMeshes(const smtk::mesh::MeshSet& meshes,
...
+    {
+    // TODO: should we clear everything and make this an invalid collection (how?)
+    return false;
+    }
+
+  smtk::model::Model inModel(this->m_modelManager.lock(), uuid);
+  // TODO: what if this->m_modelEntity is valid, should we clear everything (how?)
+  if(inModel.isValid() && this->m_modelEntity.isNull())
+    {
+    this->m_modelEntity = uuid;
```
Comment (robertmaynard) [inline smtk/mesh/Collection.cxx]: I wouldn't store m_modelEntity, instead use getModelEntity() on the mesh_iface to get the uuid.

Comment (yumin) [inline smtk/mesh/Collection.cxx]: There is not getModeleEntity() on mesh_iface. However, I could add one. I sent you an email offline.

**Comment by yumin** [comment]
This is replaced with [PR#293] (https://github.com/Kitware/SMTK/pull/293)

**Pull request closed without merging**